modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala

```diff
@@ -1275,21 +1275,58 @@ private[smithy4s] class SchemaVisitorJCodec(
         }
     }
 
-  private def jsonLabel[A, Z](field: Field[Z, A]): String =
+  private def isForUnknownFieldRetention[Z](field: Field[Z, _]): Boolean =
+    field.hints
+      .has(UnknownJsonFieldRetention)
+
+  private def jsonLabelOrLabel[Z](field: Field[Z, _]): String =
     field.hints.get(JsonName) match {
       case None    => field.label
       case Some(x) => x.value
     }
 
+  trait UnknownFieldsDecoder[A] { self =>
```

Comment:
rather than creating a bespoke trait (which implies having to create a schema visitor for it), I'd just delegate to `Document.Decoder`, turning the `Hashmap` into a `Document.DObject`

Reply:
Done in https://github.com/disneystreaming/smithy4s/commit/59e0d225da28823f1679bc8b7243e0deab05a75a.